sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -166,6 +166,42 @@ class InMemoryCatalogedDDLSuite extends DDLSuite with SharedSQLContext with Befo
     }
   }
 
+  Seq("parquet", "json", "csv").foreach { provider =>
+    test(s"Alter table add columns -- ${provider} format") {
+      assume(spark.sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "in-memory")
+      withTable("t") {
+        sql(s"create table t (c1 int) using ${provider}")
+        sql("insert into table t values (1)")
+        sql("alter table t add columns (c2 int)")
+        checkAnswer(sql("select * from t"), Seq(Row(1, null)))
+        sql("insert into table t values (2, 2)")
+        checkAnswer(sql("select * from t where c2 is not null"), Seq(Row(2, 2)))
+        checkAnswer(
+          sql("select * from t"),
+          Seq(Row(1, null), Row(2, 2))
+        )
+      }
+    }
+  }
+
+  Seq("parquet", "json", "csv").foreach { provider =>
+    test(s"Alter table add columns with partitions -- ${provider} format") {
+      assume(spark.sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "in-memory")
+      withTable("t") {
+        sql(s"create table t (c1 int, c2 int) using ${provider} partitioned by (c2)")
+        sql("insert into table t values (1, 1)")
+        sql("alter table t add columns (c3 int)")
+        checkAnswer(sql("select * from t"), Seq(Row(1, null, 1)))
+        sql("insert into table t values (2, 2, 3)")
+        checkAnswer(sql("select * from t where c3 is not null"), Seq(Row(2, 2, 3)))
+        checkAnswer(sql("select * from t where c2 = 3"), Seq(Row(2, 2, 3)))
+        checkAnswer(
+          sql("select * from t"),
+          Seq(Row(1, null, 1), Row(2, 2, 3))
+        )
+      }
+    }
+  }
```

Comment:
Why we need these two test cases? We already have them in `DDLSuite`, which is extended by `InMemoryCatalogedDDLSuite`.